examples/0-abstraction-hh-rlhf/rl_finetuning.py

```diff
@@ -0,0 +1,128 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from datasets import load_dataset
+from peft import LoraConfig
+from tqdm import tqdm
+from transformers import LlamaTokenizer
+
+from trl import AutoModelForCausalLMWithValueHead, PPOConfig, PPOTrainer
+from trl.core import LengthSampler
+
+
+rm_adapter_id = "ybelkada/llama-7b-dummy-rm"
+model_name = "decapoda-research/llama-7b-hf"
+dataset_name = "Anthropic/hh-rlhf"
+
+input_min_text_length = 6
+input_max_text_length = 12
+
+DEFAULT_PAD_TOKEN = "[PAD]"
+DEFAULT_EOS_TOKEN = "</s>"
+DEFAULT_BOS_TOKEN = "</s>"
+DEFAULT_UNK_TOKEN = "</s>"
+
+
+def create_and_prepare_dataset(tokenizer):
+    dataset = load_dataset(dataset_name, split="train[:1%]")
+
+    input_size = LengthSampler(input_min_text_length, input_max_text_length)
+
+    def tokenize(example):
+        text_size = input_size()
+        example["input_ids"] = tokenizer.encode(example["chosen"])[:text_size]
+        example["query"] = tokenizer.decode(example["input_ids"])
+        return example
+
+    dataset = dataset.map(tokenize, batched=False)
+    dataset.set_format("torch")
+    return dataset
+
+
+lora_config = LoraConfig(
+    r=16,
+    lora_alpha=32,
+    lora_dropout=0.05,
+    bias="none",
+    task_type="CAUSAL_LM",
+)
+model = AutoModelForCausalLMWithValueHead.from_pretrained(
+    model_name,
+    load_in_8bit=True,
+    device_map={"": 0},
+    peft_config=lora_config,
+    reward_modeling_adapter_id=rm_adapter_id,
+)
+tokenizer = LlamaTokenizer.from_pretrained(model_name)
+
+tokenizer.add_special_tokens(
```

Comment:
Same comment here about not needing this anymore (it might depend on the pretrained checkpoint you're using though!)